src/modules/ledger/RadixNEDBAtomStore.ts

```diff
@@ -0,0 +1,254 @@
+            })
+
+            return true
+        }).catch(error => {
+            return false
+        })
+    }    
+    
+    public updateStatus(aid: RadixAID, status: RadixAtomNodeStatusUpdate): Promise<boolean> {
+        return this.dbFindOne(aid.toString()).then(atomEntry => {
+            if (atomEntry.status.status === status.status) {
+                throw new Error('No change')
+            }
+
+            this.dbUpdate(aid.toString(), status).then(() => {
+                this.atomObservationSubject.next({
+                    atom: RadixSerializer.fromJSON(atomEntry.atom),
+                    status,
+                    timestamp: atomEntry.updatedAt,
+                })
+            })
+
+            return true
+        })
+        .catch(error => {
+            return false
+        })
+
+    }
+
+
+    public getAtom(aid: RadixAID): Promise<RadixAtomObservation> {
+        return this.dbFindOne(aid.toString()).then(atomEntry => {
+            return {
+                atom: RadixSerializer.fromJSON(atomEntry.atom),
+                status: atomEntry.status,
+                timestamp: atomEntry.updatedAt,
+            }
+        })
+    }
+
+
+    public getAtomObservations(address?: RadixAddress): Observable<RadixAtomObservation> {
+        return this.atomObservationSubject.filter(atomObservation => {
+            if (address) {
+                for (const atomAddress of atomObservation.atom.getAddresses()) {
+                    if (address.equals(atomAddress)) {
+                        return true
+                    }
+                }
+
+                return false
+            } else {
+                return true
+            }
+        }).share()
+    }
+
+
+    public getAtomStatusUpdates(aid: RadixAID): Observable<RadixAtomNodeStatusUpdate> {
+        return this.atomObservationSubject.filter((atomObservation) => {
+            return atomObservation.atom.getAid().equals(aid)
+        }).map(atomObservation => {
+            return atomObservation.status
+        }).share()
+    }
+    
+
+    public getStoredAtomObservations(address?: RadixAddress): Observable<RadixAtomObservation> {
+        const query = {}
+
+        if (address) {
+            (query as any).addresses = address.toString()
+        }
+
+        return Observable.create(observer => {
+            this.dbFind(query).then(async entires => {
+                for (const entry of entires) {
+                    const deserializedAtom = await this.asyncDeserialize(entry.atom)
+    
+                    observer.next({
+                        atom: deserializedAtom,
+                        status: entry.status,
+                        timestamp: entry.updatedAt,
+                    })
+                }
+
+                observer.complete()
+            })
+        })
+    }
+
+
+    private asyncDeserialize(atom) {
+        return new Promise<RadixAtom>((resolve, reject) => {
+            setTimeout(() => {
```

Comment:
Out of curiosity: Why this zero-delay timeout? Is this a commonly known magic trick, or would a describing comment be justifiable here?

Reply:
This is a well known trick, to understand how it works you need to know about the JS event loop https://developer.mozilla.org/en-US/docs/Web/JavaScript/EventLoop

TL;DR JS is single threaded, using a 0 timeout will schedule the function to be executed during the next "available opportunity", unblocking the current thread so that UI can render smoothly etc.

In this case, I don't think it can be quickly describe in a comment, and it's used in multiple places, I think the function name itself captures the essence of what's happening